### PR TITLE
Enable testing of group algorithms with different types for DPC++

### PR DESCRIPTION
**Pull request by jzc**
Depends on https://github.com/KhronosGroup/SYCL-CTS/pull/612, https://github.com/intel/llvm/pull/9154

### CODE REVIEW

**Review by keryell — APPROVED** [review]
Thanks.

**Comment by tomdeakin** [comment]
CTS CI needs updating. __If__ that then passes the tests, ready to merge.

File: tests/group_functions/group_reduce_fp16.cpp [hunk 1]
```diff
@@ -172,10 +160,9 @@ TEMPLATE_LIST_TEST_CASE("Group and sub-group reduce functions with init",
   return;
 #else
   if (queue.get_device().has(sycl::aspect::fp16)) {
-    // FIXME: DPCPP and ComputeCpp cannot handle cases of different types
+    // FIXME: ComputeCpp cannot handle cases of different types
     // Link to issue https://github.com/intel/llvm/issues/8341
```
Comment (bader) [inline tests/group_functions/group_reduce_fp16.cpp:164]:
```suggestion
```

File: tests/group_functions/group_scan_fp64.cpp [hunk 2]
```diff
@@ -187,10 +168,8 @@ DISABLED_FOR_TEST_CASE(hipSYCL)
 #else
   auto queue = sycl_cts::util::get_cts_object::queue();
   if (queue.get_device().has(sycl::aspect::fp64)) {
-    // FIXME: DPCPP and ComputeCpp cannot handle cases of different types
-    // Link to issue https://github.com/intel/llvm/issues/8341
-#if defined(SYCL_CTS_COMPILING_WITH_DPCPP) || \
-    defined(SYCL_CTS_COMPILING_WITH_COMPUTECPP)
+    // FIXME: and ComputeCpp cannot handle cases of different types
```
Comment (bader) [inline tests/group_functions/group_scan_fp64.cpp:171]:
```suggestion
    // FIXME: ComputeCpp cannot handle cases of different types
```

**Review by bader — APPROVED** [review]

**Pull request merged**